File: src/main/java/nz/co/searchwellington/model/WebsiteImpl.java
package nz.co.searchwellington.model;

import java.util.Date;
import java.util.Set;

public class WebsiteImpl extends ResourceImpl implements Website {
	
	private static final long serialVersionUID = 1L;
	
	Set <Feed> feeds;
    Set <Watchlist> watchlist;

    public WebsiteImpl() {        
    }
    
	public WebsiteImpl(int id, String name, String url, Date date, String description, Set<Feed> feed, Set<Watchlist> watchlist, Set<DiscoveredFeed> discoveredFeeds) {
        this.id = id;
		this.name = name;
        this.url = url;
        this.date = date;
        this.description = description;   
        this.feeds = feed;
        this.watchlist = watchlist;
        this.discoveredFeeds = discoveredFeeds;
	}
	
    public Set<Feed> getFeeds() {
        return feeds;
    }
    
    public void setFeeds(Set<Feed> feeds) {
		this.feeds = feeds;
	}

	public Set<Watchlist> getWatchlist() {
        return watchlist;
    }
    
    public String getType() {
        return "W";
     }
  
    public void setWatchlist(Set<Watchlist> watchlist) {
        this.watchlist = watchlist;
    }

}

File: src/main/scala/nz/co/searchwellington/model/WebsiteImpl.scala
package nz.co.searchwellington.model

import java.util.Date

case class WebsiteImpl(override var id: Int = 0,
                       override var `type`: String = "",
                       override var title: Option[String] = None,
                       override var description: Option[String] = None,
                       override var page: Option[String] = None,
                       override var http_status: Int = 0,
                       override var date2: Option[Date] = None,
                       override var last_scanned2: Option[Date] = None,
                       override var last_changed2: Option[Date] = None,
                       override var live_time2: Option[Date] = None,
                       override var embargoed_until2: Option[Date] = None,
                       override var held2: Boolean = true,
                       override var url_words: Option[String] = None,
                       override var geocode: Option[Int] = None,
                       override var owner: Option[Int] = None
                       // var feeds: Set[Feed] = Set(),
                       // var watchlist: Set[Watchlist] = Set()
                      ) extends Website {

  def getFeeds: Set[Feed] = Set() // TODO
  def setFeeds(feeds: Set[Feed]): Unit = {} // TODO

  override def getType: String = "W"

  def getWatchlist: Set[Watchlist] = Set()  // TODO
  def setWatchlist(watchlist: Set[Watchlist]): Unit = {}  // TODO

}
